src/foundry/client/pixi/placeables/token.d.ts

```diff
@@ -164,30 +210,35 @@ declare global {
     get emitsLight(): boolean;
 
     /**
-     * Test whether the Token has a limited angle of vision or light emission which would require sight to update on Token rotation
+     * Test whether the Token uses a limited angle of vision or light emission.
      */
-    get hasLimitedVisionAngle(): boolean;
+    get hasLimitedSourceAngle(): boolean;
 
     /**
-     * Translate the token's sight distance in units into a radius in pixels.
-     * @returns The sight radius in pixels
+     * Translate the token's dim light distance in units into a radius in pixels.
      */
     get dimRadius(): number;
 
     /**
      * Translate the token's bright light distance in units into a radius in pixels.
-     * @returns The bright radius in pixels
      */
     get brightRadius(): number;
 
     /**
-     * The named identified for the source object associated with this Token
+     * Translate the token's vision range in units into a radius in pixels.
+     */
+    get sightRange(): number;
+
+    /**
+     * Translate the token's maximum vision range that takes into account lights.
      */
-    get sourceId(): string;
+    get optimalSightRange(): number;
+
+    override clone(): ThisType<Token>;
```

Comment:
Asfik the only reason to use `ThisType` is as part of an intersection. You can see it in the example TypeScript gives:
```ts
type ObjectDescriptor<D, M> = {
  data?: D;
  methods?: M & ThisType<D & M>; // Type of 'this' in methods is D & M
};
 
function makeObject<D, M>(desc: ObjectDescriptor<D, M>): D & M {
  let data: object = desc.data || {};
  let methods: object = desc.methods || {};
  return { ...data, ...methods } as D & M;
}
 
let obj = makeObject({
  data: { x: 0, y: 0 },
  methods: {
    moveBy(dx: number, dy: number) {
      this.x += dx; // Strongly typed this
      this.y += dy; // Strongly typed this
    },
  },
});
 
obj.x = 10;
obj.y = 20;
obj.moveBy(5, 5);
```

If you leave `M & ThisType<D & M>` as `M` things break, the `ThisType` basically just sets `this` inside `M`. It's not going to be helpful here.

Reply:
You have to jump through a lot of hoops to even find a reason to use `ThisType`, It's basically for this pretty niche case.